File: performance-comparision/pages/login_page.rb
require File.join(File.dirname(__FILE__), "abstract_page.rb")

class LoginPage < AbstractPage
  def initialize(driver)
    super(driver, "") # <= TEXT UNIQUE TO THIS PAGE
  end

  def check_remember_me
    driver.find_element(:id, "remember_me").click
  end

  def enter_username(user)
    driver.find_element(:id, "username").send_keys(user)
  end

  def enter_password(passwd)
    driver.find_element(:id, "password").send_keys(passwd)
  end

  def click_sign_in
    driver.find_element(:id, "username").submit
  end
end

File: performance-comparision/pages/login_page.js
const AbstractPage = require('./abstract_page');

class LoginPage extends AbstractPage {

  constructor(driver) {
    super(driver);
  }

  async selectTripType(trip_type) {
    const radios = await this.driver.$$("input[name=tripType]");
    if (trip_type == "oneway") {
  await radios[1].check();
    } else  {
        await radios[0].check();
    }
  }

  async enterUsername(username) {
    await this.driver.fill("#username", username);
  }

  async enterPassword(passwd) {
    await this.driver.fill("#password", passwd);
  }

  async checkRememberMe() {
    await this.driver.click("#remember_me");
  }

  async clickSignIn() {
     await this.driver.click("input:has-text('Sign in')");
  }

};

module.exports = LoginPage;
